metricbeat/module/system/process/process.go

```diff
@@ -111,7 +112,8 @@ func (m *MetricSet) Fetch(r mb.ReporterV2) error {
 	// monitor either a single PID, or the configured set of processes.
 	if m.setpid == 0 {
 		procs, roots, err := m.stats.Get()
-		if err != nil {
+		if err != nil && !errors.Is(err, process.NonFatalErr{}) {
```

Comment:
Non-fatal errors should probably set the `Error` field of the event: https://github.com/elastic/beats/blob/032a4cfd5f3b8fa8354ac1e0062a0e1f196c60d0/metricbeat/mb/event.go#L44

Reply:
@cmacknz If you have a look https://github.com/elastic/beats/blob/6e88ac5c0335822d0b2c1fc2eccc96213747b365/metricbeat/mb/module/wrapper.go#L263 this `Error()` call would result in an event with an `Error: err` set, it happens https://github.com/elastic/beats/blob/6e88ac5c0335822d0b2c1fc2eccc96213747b365/metricbeat/mb/module/wrapper.go#L357

Also, I've verified with test cases that the error events are indeed generated.